ADMVEDU95/ADMVEDU95/ViewControllers/CategoryViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  CategoryViewController.swift
+//  ADMVEDU95
+//
+//  Created by Satsishur on 16.04.2021.
+//
+
+import UIKit
+
+protocol CategoryDelegate {
+    func fetchCategory(category: iTunesCategory)
+}
+
+class CategoryViewController: UIViewController {
+    struct Constants {
+        static let categoryCellIdentifier = "CategoryTableViewCell"
+    }
+    
+    var dataSource: [iTunesCategory] = {
+        var data = [iTunesCategory]()
+        for category in iTunesCategory.allCases {
+            data.append(category)
+        }
+        return data
+    }()
+    
+    var delegate: CategoryDelegate?
+    
+    var categoryChosed: iTunesCategory?
+    
+    private var tableView: UITableView = {
+        let tableView = UITableView()
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: Constants.categoryCellIdentifier)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        return tableView
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableViewLayout()
+    }
+    
+    private func setupTableViewLayout() {
+        view.addSubview(tableView)
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
```

Comment:
there are a lot of similar code to configure constraints - let's add functions to UIView extensions to configure it